tools/run_tests/helper_scripts/build_python.sh

```diff
@@ -157,11 +157,10 @@ curl https://bootstrap.pypa.io/get-pip.py | $VENV_PYTHON
 # pip-installs the directory specified. Used because on MSYS the vanilla Windows
 # Python gets confused when parsing paths.
 pip_install_dir() {
-  PWD=$(pwd)
-  cd "$1"
+  pushd "$1"
```

Comment:
The proper way to do this in bash is the following:

```(
cd "$1"
#code
# no cd needed; subshell ends
)```

Reply:
I actually considered using it here, but I wasn't sure if un`export`ed variables would propagate to the child. Glad to see that it does work.

I like the lack of explicit change to the original directory, but the fact that it requires a new subshell isn't ideal for all use cases. I still don't think you could, for example, modify a variable in the parent shell.